---
---
title: "Masters-phyloseqs"
output: html_document
---
```{r}
library(tidyverse)
library(tidyr)
library(RAM)
library(rstatix)
library(devtools)
library(phyloseq)
library(dada2)
library(plyr)
library(dplyr)
library(microbiome)
library(vegan)
library(ggplot2)
library(gtools)
library(data.table)
library(rcompanion)
library(e1071)
library(ggpubr)
library(DESeq2)
library(meta)
library(fdrtool)
library(RColorBrewer)
library("genefilter")
library(SummarizedExperiment)
library(hrbrthemes)
library(forcats)
library(RColorBrewer)
library(apeglm)
library(fastDummies)
library(hablar)
library(EnvStats)
library(effsize)
library(scales)
library(patchwork)
```


Yamada
```{r}
setwd("Data/Yamada/")
sequ_yam<-readRDS("Data/Yamada/seqtab_final.rds")
taxa_yam<-readRDS("Data/Yamada/tax_final.rds")
meta_yam<-read.csv("Data/Yamada/yamada_meta.csv")
```


```{r}
samples.out<-rownames(sequ_yam)
subject<-samples.out
sampledf_yam<-data.frame(Subject=subject)
```

Rarefy
```{r}
#Making all dataframes have the same codes for each tested group
colnames(meta_yam)[1]<- "Subject"
meta_yam$dx<-ifelse(grepl('^[HIK]', meta_yam$host_subject_id), "nonIBD", 
                    ifelse(grepl('^U',  meta_yam$host_subject_id),"UC", 
                          ifelse(grepl('^C',  meta_yam$host_subject_id),"CD","other" )))
samdf_yam<-merge(sampledf_yam, meta_yam, by = "Subject")
rownames(samdf_yam) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_yam<- phyloseq(otu_table(sequ_yam, taxa_are_rows = FALSE),
                  sample_data(samdf_yam),
                  tax_table(taxa_yam))

#make it relative abundance
#ps.tx.jac<-transform_sample_counts(ps_jac, function(OTU) OTU/sum(OTU))
```

Rarefy 
```{r}
prune_ps_yam<-prune_samples(sample_sums(ps_yam)>=1000, ps_yam)

rarex = rarefy_even_depth(rngseed =1, prune_ps_yam, 1000)

rare_yam<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_yam)
ntaxa(rare_yam)
nsamples(ps_yam)
ntaxa(ps_yam)
table(sample_data(ps_yam)$dx)
table(sample_data(rare_yam)$dx)
```

mott
```{r}
setwd("Data/Mottawea/")
sequ_mott<-readRDS("Data/Mottawea/seqtab_final.rds")
taxa_mott<-readRDS("Data/Mottawea/tax_final.rds")
meta_mott<-read.csv("Data/Mottawea/meta_mott.csv")
```


```{r}
samples.out<-rownames(sequ_mott)
subject<-samples.out
sampledf_mott<-data.frame(Subject=subject)
```

Rarefy
```{r}
meta_mott$dx<-ifelse(meta_mott$disease =="Control", "nonIBD",
                     ifelse(meta_mott$disease=="CD", "CD", "UC"))

colnames(meta_mott)[1]<-"Subject"
samdf_mott<-merge(sampledf_mott, meta_mott, by = "Subject")
rownames(samdf_mott) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_mott<- phyloseq(otu_table(sequ_mott, taxa_are_rows = FALSE),
                  sample_data(samdf_mott),
                  tax_table(taxa_mott))

```

Rarefy 
```{r}
prune_ps_mott<-prune_samples(sample_sums(ps_mott)>=10000, ps_mott)

rarex = rarefy_even_depth(rngseed =1, prune_ps_mott, 10000)

rare_mott<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_mott)
ntaxa(rare_mott)
nsamples(ps_mott)
ntaxa(ps_mott)
table(sample_data(ps_mott)$dx)
table(sample_data(rare_mott)$dx)
```

Liu
```{r}
setwd("Data/Liu/")
sequ_liu<-readRDS("Data/Liu/seqtab_final.rds")
taxa_liu<-readRDS("Data/Liu/tax_final.rds")
meta_liu<-read.csv("Data/Liu/meta_liu.csv")
```


```{r}
samples.out<-rownames(sequ_liu)
subject<-samples.out
sampledf_liu<-data.frame(Subject=subject)
```

Rarefy
```{r}
colnames(meta_liu)[1]<- "Subject"
colnames(meta_liu)[40]<-"Subject2"
meta_liu$dx<-ifelse(meta_liu$Diagnosis=="Not IBD", "nonIBD", "CD")
samdf_liu<-merge(meta_liu, sampledf_liu, by = "Subject")
rownames(samdf_liu) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_liu<- phyloseq(otu_table(sequ_liu, taxa_are_rows = FALSE),
                  sample_data(samdf_liu),
                  tax_table(taxa_liu))

```

Rarefy 
```{r}
prune_ps_liu<-prune_samples(sample_sums(ps_liu)>=10000, ps_liu)

rarex = rarefy_even_depth(rngseed =1, prune_ps_liu, 4000)

rare_liu<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_liu)
ntaxa(rare_liu)
nsamples(ps_liu)
ntaxa(ps_liu)
table(sample_data(ps_liu)$dx)
table(sample_data(rare_liu)$dx)
```

Braun
```{r}
setwd("Data/Braun/")
sequ_br<-readRDS("Data/Braun/seqtab_final.rds")
taxa_br<-readRDS("Data/Braun/tax_final.rds")
meta_br<-read.csv("Data/Braun/meta_br.csv")
```


```{r}
samples.out<-rownames(sequ_br)
subject<-samples.out
sampledf_braun<-data.frame(Subject=subject)
```

Rarefy
```{r}
colnames(meta_br)[1]<- "Subject"

```

Making it into a phyloseq object 

```{r}
ps_br<- phyloseq(otu_table(sequ_br, taxa_are_rows = FALSE),
                  sample_data(samdf_br),
                  tax_table(taxa_br))

```

Rarefy 
```{r}
prune_ps_br<-prune_samples(sample_sums(ps_br)>=4000, ps_br)

rarex = rarefy_even_depth(rngseed =1, prune_ps_br, 4000)

rare_br<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_br)
ntaxa(rare_br)
nsamples(ps_br)
ntaxa(ps_br)
table(sample_data(ps_br)$dx)
table(sample_data(rare_br)$dx)

```



PROTECT
```{r}
setwd("Data/Schirmer/")
sequ_pro<-readRDS("Data/Schirmer/seqtab_final.rds")
taxa_pro<-readRDS("Data/Schirmer/tax_final.rds")
list_pro<-read.csv("Data/Schirmer/protect.csv")
meta_pro<-read.csv("Data/Schirmer/mmc7.csv")
```
merging meta and list 
```{r}
colnames(meta_pro)[1]<-"Sample.Name"
meta_pro<-meta_pro[,1:24] #take out the columns of value
meta_pro$Sample.Name<-as.factor(as.character(meta_pro$Sample.Name))
meta_pro<-meta_pro %>% 
  arrange(Sample.Name) #arrange sample names to match other file so that merging can happen effortlessly 


list_pro$Sample.Name<-as.factor(as.character(list_pro$Sample.Name))
list_pro<-list_pro %>% 
  arrange(Sample.Name)
samdf_pro<-merge(list_pro,meta_pro, by = "Sample.Name")
samdf_pro$Inflamed<-ifelse(samdf_pro$PUCAI_C4_WKall=="inactive", "0","1") #controls are inactive UC. Change identifiable code to match other dtataframes for comparison 
samdf_pro$dis<-ifelse(samdf_pro$Inflamed=="0", "nonIBD", "UC")
```

```{r}
samples.out<-rownames(sequ_pro)
subject<-samples.out
sampledf_pro<-data.frame(Subject=subject)
```



Rarefy
```{r}
colnames(samdf_pro)[2]<- "Subject"

samdf_pro<-merge(sampledf_pro, samdf_pro, by = "Subject")
rownames(samdf_pro) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_pro<- phyloseq(otu_table(sequ_pro, taxa_are_rows = FALSE),
                  sample_data(samdf_pro),
                  tax_table(taxa_pro))

#make it relative abundance
#ps.tx.jac<-transform_sample_counts(ps_jac, function(OTU) OTU/sum(OTU))
```

 Rarefy 
```{r}
prune_ps_pro<-prune_samples(sample_sums(ps_pro)>=1212, ps_pro)

rarex = rarefy_even_depth(rngseed =1, prune_ps_pro, 1000)

rare_pro<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_pro)
ntaxa(rare_pro)
nsamples(ps_pro)
ntaxa(ps_pro)
```


XCM
```{r}
setwd("Data/Morgan_2012/")
sequ_xcm<-readRDS("Data/Morgan_2012/seqtab_final.rds")
taxa_xcm<-readRDS("Data/Morgan_2012/tax_final2.rds")
meta_xcm<-read.csv("Data/Morgan_2012/osccar_prism_metadata.csv")
list_xcm<-read.csv("Data/Morgan_2012/sra_run.csv")
```

```{r}
samples.out<-rownames(sequ_xcm)
subject<-samples.out
sampledf_xcm<-data.frame(Subject=subject)
```
metadata clean up
```{r}
#clean up structure of metadata (2 files to fit together) to fit with sequencing data 
table<-as.data.frame(t(meta_xcm))
colnames(table)<-meta_xcm[,1]
table_xcm<-tibble::rownames_to_column(table, "Sample")
xcm_df<-table_xcm[-c(1),]
xcm_df$Sample<-gsub("[X]","", xcm_df$Sample )
colnames(list_xcm)[42]<-"Sample"

meta_data<-merge(xcm_df, list_xcm, by= "Sample", all.x=TRUE)

colnames(meta_data)[20]<- "Subject"
meta_data$dx<-gsub("Healthy","nonIBD", meta_data$dx)

sampledf_xcm<-subset(sampledf_xcm, Subject %in% meta_data$Subject)
samdf_xcm<-merge(sampledf_xcm, meta_data, by = "Subject")

sam<-samples.out[1:228]
rownames(samdf_xcm) <- sam

```

Making it into a phyloseq object 

```{r}
ps_xcm<- phyloseq(otu_table(sequ_xcm, taxa_are_rows = FALSE),
                  sample_data(samdf_xcm),
                  tax_table(taxa_xcm))

```
 Rarefy 
```{r}
prune_ps_xcm<-prune_samples(sample_sums(ps_xcm)>=100, ps_xcm)

rarex = rarefy_even_depth(rngseed =1, prune_ps_xcm, 900)

rare_xcm<- prune_taxa(taxa_sums(rarex)>1, rarex)
nsamples(rare_xcm)
ntaxa(rare_xcm)
nsamples(ps_xcm)
ntaxa(ps_xcm)
```

```{r}
setwd("Data/Gevers")

```

```{r}
sequ_risk<-readRDS("Data/Gevers/seqtab_final.rds")
taxa_risk<-readRDS("Data/Gevers/tax_final.rds")
meta_risk<-read.csv("Data/Gevers/SraRunTable.csv")
map_risk<-read.csv("Data/Gevers/risk_mapping_file.csv")

map_risk2<-read.csv("Data/Gevers/mapping_risk_2.csv")
meta_bio<-read.csv("Data/Gevers/meta_bio.csv")


sequ_risk_stool<-readRDS("Data/Gevers/seqtab_final_stool.rds")
taxa_risk_stool<-readRDS("Data/Gevers/tax_final_stool.rds")
map_stool<-read.csv("Data/Gevers/risk_stool_meta.csv")
```

```{r}
samples.out<-rownames(sequ_risk)
subject<-samples.out
sampledf<-data.frame(Subject =subject)
```

Filtering to only have samples we picked for this project
```{r}
map_risk$external_id<-gsub("[.]","", map_risk$external_id ) # remove . or wildcard to share same code with other file
map_risk2$external_id<-gsub("[.]","", map_risk2$external_id ) # remove . or wildcard to share same code with other file
meta_risk$Sample.Name<-gsub("-","", meta_risk$Sample.Name )

meta_bio<-meta_bio[!meta_bio$biopsy_location == "stool",]

meta_bio$run_prefix..exp.<-gsub(".*[.]", "", meta_bio$run_prefix..exp.) #run prefix is shared between two files

map_risk2$run_prefix<-gsub(".*[.]", "", map_risk2$run_prefix)#run prefix is shared between two files

meta_filtered<- meta_risk %>%
  filter(Subject %in% sampledf$Subject) # filter out samples that you do not have the meta-data for. 


meta_bio$external_id..exp.<-gsub("[.]","", meta_bio$external_id..exp. ) #Help patterns match so that mapping files cna merge succesfully 

meta_bio<- meta_bio%>%
   filter( meta_bio$external_id..exp.%in% meta_filtered$Sample.Name)
  
x<-meta_filtered %>% 
  find_duplicates(Library.Name, Sample.Name) #remove duplicated sample names so that you only have one sample from each patient

y<-meta_bio %>% 
  find_duplicates(run_prefix..exp.,external_id..exp.)

meta_filtered1<-meta_filtered[!meta_filtered$Subject %in% x$Subject,]

colnames(meta_bio)[46]<-"Library.Name"

colnames(meta_bio)[46]<-"Sample.Name"
colnames(meta_bio)[76]<-"yy"

new<-mutate(meta_bio, Sample.Name = as.character(Sample.Name)) %>% 
                  left_join(meta_filtered1, ., by = 'Sample.Name')
dif<-setdiff(meta_filtered$Subject, new$Subject)

toadd<-meta_filtered[meta_filtered$Subject %in% dif,]

meta<-rbind.fill(new, toadd)

z<-meta$Subject[duplicated(meta$Subject)]



meta_filtered<-meta%>%
  group_by(Subject) %>% 
  filter(row_number()==1) 


#substitute variable names so that it is uniform for all datasets

meta_filtered$lat_lon<-gsub("Chrons Disease", "CD", meta_filtered$lat_lon)
meta_filtered$lat_lon<-gsub("Crohn's Disease", "CD", meta_filtered$lat_lon)
meta_filtered$lat_lon<-gsub("Not IBD","nonIBD", meta_filtered$lat_lon)
meta_filtered$lat_lon<-gsub("Control","nonIBD", meta_filtered$lat_lon)
 meta_filtered$lat_lon<-gsub("Ulcerative Colitis", "UC", meta_filtered$lat_lon)
 meta_filtered$lat_lon<-gsub("IBD-U", "UC", meta_filtered$lat_lon)
```



Merging metadata and sequence table
```{r}

colnames(samdf_risk)[65]<-"dx"
rownames(samdf_risk) <- samples.out 

```

Making it into a phyloseq object 
```{r}
ps_risk<- phyloseq(otu_table(sequ_risk, taxa_are_rows = FALSE),
                   sample_data(samdf_risk),
                   tax_table(taxa_risk))


```

Risk stool
```{r}
#repeat the same process for Gevers biopsy dataset to teh stool dataset 

samples.out<-rownames(sequ_risk_stool) #279
subject<-samples.out
sampledf<-data.frame(Subject =subject)

colnames(map_stool)[1]<- "Subject"


map_risk2$run_prefix<-gsub(".*[.]", "", map_risk2$run_prefix)
map_stool$run_prefix..exp.<-gsub(".*[.]", "", map_stool$run_prefix..exp.)

colnames(map_stool)[80]<-"run_prefix"

new<-mutate(map_risk2,  run_prefix = as.character(run_prefix)) %>% 
                  left_join(map_stool, ., by = 'run_prefix')

samdf_risk_stool<-merge(x=sampledf, y= new, by = "Subject")

colnames(samdf_risk_stool)[65]<-"dx"
samdf_risk_stool<-samdf_risk_stool[samdf_risk_stool$Subject %in% samples.out,]
rownames(samdf_risk_stool) <- samples.out 

colnames(samdf_risk_stool)[33]<-"diagnosis"

samdf_risk_stool$diagnosis<-gsub("no","nonIBD", samdf_risk_stool$diagnosis)

```
Phyloseq object
```{r}
ps_risk_stool<- phyloseq(otu_table(sequ_risk_stool, taxa_are_rows = FALSE),
                  sample_data(samdf_risk_stool),
                  tax_table(taxa_risk_stool))
```

biopsy-rarefy
```{r}
prune_ps_risk<-prune_samples(sample_sums(ps_risk)>1000, ps_risk)
rare = rarefy_even_depth(rngseed =1, prune_ps_risk, 10000)
rare_risk<- prune_taxa(taxa_sums(rare)>1, rare)
nsamples(rare_risk)
nsamples(ps_risk)
ntaxa(ps_risk)
```
Stool-rarefy
```{r}
prune_ps_risk<-prune_samples(sample_sums(ps_risk_stool)>1000, ps_risk_stool)
rare = rarefy_even_depth(rngseed =1, prune_ps_risk, 10000)
rare_risk_stool<- prune_taxa(taxa_sums(rare)>1, rare)
nsamples(rare_risk_stool)
nsamples(ps_risk_stool)
ntaxa(ps_risk_stool)
```


HMP study
```{r}
setwd("Data/Lloyd_Price/")
sequ_hmp<-readRDS("Data/Lloyd_Price/seqtab_final.rds")
taxa_hmp<-readRDS("Data/Lloyd_Price/tax_final.rds")
meta_hmp<-read.csv("Data/Lloyd_Price/SraRunTable.txt")
meta_hmp2<-read.csv("Data/Lloyd_Price/hmp2_metadata.csv")
```

```{r}
samples.out<-rownames(sequ_hmp)
subject<-samples.out
sampledf_hmp<-data.frame(Subject =subject)
```

Merging metadata and sequence table
```{r}
colnames(meta_hmp)[1]<- "Subject"

names(meta_hmp)[37]<- "dx"

meta_hmp2<-meta_hmp2[meta_hmp2$External.ID %in% meta_hmp$external_sample_id,] #select for samples you have the appropriate meta-data for to allow for analysis 
colnames(meta_hmp2)[2]<-"external_sample_id"

meta_hmp_allll<-merge(meta_hmp, meta_hmp2, by ="external_sample_id")
meta_hmp_both<-meta_hmp_allll[,-c(48,52:65,71:80,83,47:91,112:132,137:258,329:334,356:457, 501:531)] #remove unnecessary columns 


meta_hmp_both<-meta_hmp_both[,-c(27,71:136)]



samdf_hmp<-merge(x=sampledf_hmp, y= meta_hmp_both, by = "Subject")


rownames(samdf_hmp) <- samples.out 
```
Making it into a phyloseq object 

```{r}
ps_hmp<- phyloseq(otu_table(sequ_hmp, taxa_are_rows = FALSE),
                  sample_data(samdf_hmp),
                  tax_table(taxa_hmp))


```


```{r}
prune_ps_hmp<-prune_samples(sample_sums(ps_hmp)>150, ps_hmp)
rareh = rarefy_even_depth(rngseed =1, prune_ps_hmp, 1700)
rare_hmp<- prune_taxa(taxa_sums(rareh)>1, rareh)
nsamples(rare_hmp)
ntaxa(rare_hmp)
nsamples(ps_hmp)
ntaxa(ps_hmp)


```

```{r}
dna <- Biostrings::DNAStringSet(taxa_names(rare_hmp))
names(dna) <- taxa_names(rare_hmp)
rare_hmp<- merge_phyloseq(rare_hmp, dna)
taxa_names(rare_hmp) <- paste0("ASV", seq(ntaxa(rare_hmp)))
```

Jansson
```{r}

setwd("Data/Halfvarson/")
sequ_jan<-readRDS("Data/Halfvarson/seqtab_final.rds")
taxa_jan<-readRDS("Data/Halfvarson/tax_final.rds")
meta_jan<-read.csv("Data/Halfvarson/Jansson_2017.csv")
list<-read.csv("Data/Halfvarson/ebi_experiment_jan2017.csv")


```

Changing files for a successful merging 
```{r}

list_jan<-list[order(list$sample_name),]#make sure sample names are in the same order in both files 
meta_jan<-meta_jan[order(meta_jan$X.SampleID),] 
colnames(meta_jan)[1]<- "experiment" #make sure column names are the same 
colnames(list_jan)[1]<- "experiment"
metadata_jan<-cbind(x=list_jan, y=meta_jan, by = "experiment")
metadata_jan$y.Subject<-NULL
colnames(metadata_jan)[1]<-"experiment"
colnames(metadata_jan)[2]<-"Subject"
```

Change names of variables
```{r}
names(metadata_jan)[28]<- "dx"
metadata_jan$dx<-gsub("HC", "nonIBD", metadata_jan$dx)

```


```{r}
samples.out<-rownames(sequ_jan)
subject<-samples.out
sampledf_jan<-data.frame(Subject=subject)
```

Rarefy
```{r}
samdf_jan<-merge(sampledf_jan, metadata_jan, by = "Subject")
samdf_jan$y.collection_timestamp <- as.Date(samdf_jan$y.collection_timestamp, "%m/%d/%Y") # Need this structure for when controlling for pseudo-replication 


rownames(samdf_jan) <- samples.out 


```

Making it into a phyloseq object 

```{r}
ps_jan<- phyloseq(otu_table(sequ_jan, taxa_are_rows = FALSE),
                  sample_data(samdf_jan),
                  tax_table(taxa_jan))

```



```{r}
prune_ps_jan<-prune_samples(sample_sums(ps_jan)>600, ps_jan) #remove samples that have less than 600 reads
rarej = rarefy_even_depth(rngseed =1, prune_ps_jan, 10000)
rare_jan<- prune_taxa(taxa_sums(rarej)>1, rarej)
nsamples(rare_jan)
ntaxa(rare_jan)
nsamples(ps_jan)
ntaxa(ps_jan)

```


Jacobs
```{r}
setwd("Data/Jacobs/")
sequ_jac<-readRDS("Data/Jacobs/seqtab_final.rds")
taxa_jac<-readRDS("Data/Jacobs/tax_final.rds")
meta_jac<-read.csv("Data/Jacobs/Jacob_meta.csv")
```

```{r}
samples.out<-rownames(sequ_jac)
subject<-samples.out
sampledf_jac<-data.frame(Subject=subject)
```

Rarefy
```{r}
colnames(meta_jac)[1]<- "Subject"
meta_jac$gastrointest_disord<-gsub("Normal","nonIBD", meta_jac$gastrointest_disord)
names(meta_jac)[19]<- "dx"
samdf_jac<-merge(sampledf_jac, meta_jac, by = "Subject")
rownames(samdf_jac) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_jac<- phyloseq(otu_table(sequ_jac, taxa_are_rows = FALSE),
                  sample_data(samdf_jac),
                  tax_table(taxa_jac))

#make it relative abundance
#ps.tx.jac<-transform_sample_counts(ps_jac, function(OTU) OTU/sum(OTU))
```



```{r}
prune_ps_jac<-prune_samples(sample_sums(ps_jac)>500, ps_jac)
rarej = rarefy_even_depth(rngseed =1, prune_ps_jac, 6000)
rare_jac<- prune_taxa(taxa_sums(rarej)>1, rarej)
nsamples(rare_jac)
ntaxa(rare_jac)
nsamples(ps_jac)
ntaxa(ps_jac)
```


Forbes
```{r}
setwd("Data/Forbes/")
sequ_forbes<-readRDS("Data/Forbes/seqtab_final.rds")
taxa_forbes<-readRDS("Data/Forbes/tax_final.rds")
meta_forbes<-read.csv("Data/Forbes/forbes_data.csv")


```

```{r}
samples.out<-rownames(sequ_forbes)
subject<-samples.out
sampledf_forbes<-data.frame(Subject=subject)
```

Rarefy
```{r}
colnames(meta_forbes)[1]<- "Subject"
colnames(meta_forbes)[21]<- "dx"
meta_forbes$dx<-gsub("HC","nonIBD", meta_forbes$dx)
meta_forbes$dx<-gsub("Crohn's disease","CD", meta_forbes$dx)
meta_forbes$dx<-gsub("Ulcerative colitis","UC", meta_forbes$dx)


samdf_forbes<-merge(sampledf_forbes, meta_forbes, by = "Subject")
rownames(samdf_forbes) <- samples.out 

```


Making it into a phyloseq object 

```{r}
ps_forbes<- phyloseq(otu_table(sequ_forbes, taxa_are_rows = FALSE),
                     sample_data(samdf_forbes),
                     tax_table(taxa_forbes))

#make it relative abundance
ps.tx.forbes<-transform_sample_counts(ps_forbes, function(OTU) OTU/sum(OTU))
```



```{r}
prune_ps_forbes<-prune_samples(sample_sums(ps_forbes)>500, ps_forbes)
rarefor = rarefy_even_depth(rngseed =1, prune_ps_forbes, 1000)
rare_forbes<- prune_taxa(taxa_sums(rarefor)>1, rarefor)
nsamples(rare_forbes)
ntaxa(rare_forbes)
nsamples(ps_forbes)
ntaxa(ps_forbes)
```

POUCH-XCMP
```{r}
setwd("Data/Morgan_2015/")
sequ_xcmp<-readRDS("Data/Morgan_2015/seqtab_final.rds")
taxa_xcmp<-readRDS("Data/Morgan_2015/tax_final.rds")
meta_xcmp<-read.csv("Data/Morgan_2015/xcm_pouch_meta.csv")
pouch_meta<-read.csv("Data/Morgan_2015/pouch_meta.csv")
```

```{r}
samples.out<-rownames(sequ_xcmp)
subject<-samples.out
sampledf_xcmp<-data.frame(Subject=subject)
sampledf_xcmp<-gsub("_1*", "", sampledf_xcmp$Subject) #change pattern in file to merge with others later
sampledf_xcmp<-as.data.frame(sampledf_xcmp)
colnames(sampledf_xcmp)[1]<-"Subject"
```

```{r}
#Change disease codes with those of other datasets for easy comparisons
meta_xcmp$dx <-ifelse(meta_xcmp$Outcome== "CDL", 'CDL',
                      ifelse(meta_xcmp$Outcome== "NP", 'UC', 
                             ifelse(meta_xcmp$Outcome=="AP", 'UC',
                                    ifelse(meta_xcmp$Outcome=="CP", 'UC','nonIBD'))))
colnames(pouch_meta)[48]<-"Sample"
colnames(meta_xcmp)[5]<-"Sample"

meta_pouch<-merge(meta_xcmp,pouch_meta, by="Sample")
meta_pouch1<-meta_pouch[,-c(31:48)] #remove unnessecary columns 
colnames(meta_pouch1)[13]<-"Subject"


samdf_xcmp<-merge(sampledf_xcmp, meta_pouch1, by = "Subject") 


rownames(samdf_xcmp) <- samples.out
```

```{r}
ps_xcmp<- phyloseq(otu_table(sequ_xcmp, taxa_are_rows = FALSE),
                   sample_data(samdf_xcmp),
                   tax_table(taxa_xcmp))

```



```{r}
prune_ps_xcmp<-prune_samples(sample_sums(ps_xcmp)>100, ps_xcmp)
rarexcmp = rarefy_even_depth(rngseed =1, prune_ps_xcmp, 500)
rare_xcmp<- prune_taxa(taxa_sums(rarexcmp)>1, rarexcmp)
rare_xcmp<-subset_samples(rare_xcmp, Outcome != "CDL")
nsamples(rare_xcmp)
ntaxa(rare_xcmp)
nsamples(ps_xcmp)
ntaxa(ps_xcmp)
```

KIM
```{r}
setwd("Data/Kim/")
sequ_kim<-readRDS("Data/Kim/seqtab_final.rds")
taxa_kim<-readRDS("Data/Kim/tax_final.rds")
meta_kim<-read.csv("Data/Kim/meta_kim.csv")
```

```{r}
samples.out<-rownames(sequ_kim)
subject<-samples.out
sampledf_kim<-data.frame(Subject=subject)
```

Rarefy
```{r}
colnames(meta_kim)[1]<- "Subject"

meta_kim$dx <-ifelse(meta_kim$feature== "Crohn's disease(stool)", 'CD',
                     ifelse(meta_kim$feature== "Crohn's disease(tissue)", 'CD', 
                            ifelse(meta_kim$feature=="health control(stool)", 'nonIBD', 'NA')))
colnames(meta_kim)[18]<-"Sample_type"
meta_kim$Study<-"Kim et al"
meta_kim$Sample_type<-gsub("^.+stool.", "stool", meta_kim$Sample_type) #change sample type code to match that of other datasets 
meta_kim$Sample_type<-gsub("^.+tissue.", "biopsy", meta_kim$Sample_type)

meta_kim$Sample_type<-as.factor(as.character(meta_kim$Sample_type))
samdf_kim<-merge(sampledf_kim, meta_kim, by = "Subject")
rownames(samdf_kim) <- samples.out 

```

Making it into a phyloseq object 

```{r}
ps_kim<- phyloseq(otu_table(sequ_kim, taxa_are_rows = FALSE),
                  sample_data(samdf_kim),
                  tax_table(taxa_kim))

#make it relative abundance
#ps.tx.jac<-transform_sample_counts(ps_jac, function(OTU) OTU/sum(OTU))
```


```{r}
sample_sums(ps_kim)
prune_ps_kim<-prune_samples(sample_sums(ps_kim)>500, ps_kim)
rarej = rarefy_even_depth(rngseed =1, prune_ps_kim, 1000)
rare_kim<- prune_taxa(taxa_sums(rarej)>1, rarej)
nsamples(rare_kim)
ntaxa(rare_kim)
nsamples(ps_kim)
ntaxa(ps_kim)
```

